Classes/Bookmark/BookmarksViewController.swift

```diff
@@ -0,0 +1,218 @@
+//
+//  BookmarksViewController.swift
+//  Freetime
+//
+//  Created by Rizwan on 18/10/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+
+class BookmarksViewController:
+    UITableViewController,
+    UISearchBarDelegate,
+    PrimaryViewController,
+TabNavRootViewControllerType {
+  
+    private let client: GithubClient
+    private let cellIdentifier = "bookmark_cell"
+    private let bookmarkStore = BookmarksStore.shared
+    private let searchBar = UISearchBar()
+    private var filterdBookmarks: [BookmarkModel]? = nil
+    
+    // MARK: Init
+    
+    init(client: GithubClient) {
+        self.client = client
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: View Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        searchBar.delegate = self
+        searchBar.placeholder = NSLocalizedString("Search", comment: "")
+        searchBar.tintColor = Styles.Colors.Blue.medium.color
+        searchBar.backgroundColor = .clear
+        searchBar.searchBarStyle = .minimal
+        searchBar.sizeToFit()
+        tableView.tableHeaderView = searchBar
+        
+        navigationItem.rightBarButtonItem = UIBarButtonItem(
+            title: NSLocalizedString("Clear All", comment: ""),
+            style: .plain,
+            target: self,
+            action: #selector(BookmarksViewController.onClearAll(sender:))
+        )
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        rz_smoothlyDeselectRows(tableView: tableView)
+        tableView.reloadData()
+        updateRightBarItem()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+    }
+    
+    // MARK: Private API
+    
+    func updateRightBarItem() {
+        navigationItem.rightBarButtonItem?.isEnabled = bookmarkStore.bookmarks.count > 0
+    }
+    
+    @objc private func onClearAll(sender: UIBarButtonItem) {
+        let alert = UIAlertController.configured(
+            title: NSLocalizedString("Bookmarks", comment: ""),
+            message: NSLocalizedString("Clear all bookmarks?", comment: ""),
+            preferredStyle: .alert
```

Comment:
Wondering if we should try match pattern with other confirmations, e.g. sign out?

"Are you sure?"
"All of your bookmarks will be lost. Do you want to continue?"